api/src/main/java/io/grpc/TimeUtils.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc;
+
+import java.time.Duration;
+
+public final class TimeUtils {
```

Comment:
Don't create a new public API. This should be package-private.

Reply:
We are using it from a different package io.grpc.stub.AbstractStub too. We'd have to make a copy of this class then. Is that what we want? @ejona86

Reply:
We have a way to expose internal classes/methods from api:
1. The class/method itself is package-private
2. Create a new class for the class name that needs exposing, prefixing the class name with `Internal` and marking it `@Internal`
3. Add accessors in that `InternalFoo` class.

So here you'd create a public `InternalTimeUtils` class that has `convert()` on it. We do the dance because 1) we hide `Internal*` classes from the javadoc, and 2) when we compile with Blaze we build the `Internal*` classes as a separate target which is visibility-restricted.

We normally do that for APIs needed to coordinate though, not utilities (core has most utilities). In this case, just duplicating the utility seems reasonable. We'll have at most three copies of it: api, stub, core. Most other things depend on core and it it can be public in core. Not the greatest, but it is small.